back-end/java/main/java/src/Parsing/RoomParser.java

```diff
@@ -0,0 +1,153 @@
+package src.Parsing;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+
+import org.apache.commons.csv.CSVRecord;
+import src.DormRoom.*;
+
+public class RoomParser {
+
+    public Parser parser;
+
+    public RoomParser(String path) throws FileNotFoundException, IOException {
+        this.parser = new Parser(path);
+    }
+
+    public ArrayList<IDormRoom> getRooms(){
```

Comment:
This method is pretty long. Please consider dividing this into more helper functions. Additionally, I see the potential for NullPointerExceptions, if the dataset was not properly constructed. While the dataset you made is beautiful and perfect ❤️ , I think a NullPointer should be handled if the dataset was accidentally tampered with.